src/Files.App/ViewModels/SidebarViewModel.cs

```diff
@@ -240,91 +143,110 @@ public SidebarViewModel()
 		{
 			if (section is null)
-			{
 				return;
-			}
 
 			switch (e.Action)
 			{
 				case NotifyCollectionChangedAction.Add:
-					{
-						for (int i = 0; i < e.NewItems.Count; i++)
-						{
-							var index = e.NewStartingIndex < 0 ? -1 : i + e.NewStartingIndex;
-							await AddElementToSection((INavigationControlItem)e.NewItems[i], section, index);
-						}
+					if (e.NewItems is null)
 						break;
+					for (int i = 0; i < e.NewItems.Count; i++)
+					{
+						var index = e.NewStartingIndex < 0 ? -1 : i + e.NewStartingIndex;
+						await AddElementToSection((INavigationControlItem)e.NewItems[i]!, section, index);
 					}
+					break;
 				case NotifyCollectionChangedAction.Move:
 				case NotifyCollectionChangedAction.Remove:
 				case NotifyCollectionChangedAction.Replace:
-					{
-						foreach (INavigationControlItem elem in e.OldItems)
-						{
-							var match = section.ChildItems.FirstOrDefault(x => x.Path == elem.Path);
-							section.ChildItems.Remove(match);
-						}
-						if (e.Action != NotifyCollectionChangedAction.Remove)
-						{
-							goto case NotifyCollectionChangedAction.Add;
-						}
+					if (e.OldItems is null)
 						break;
-					}
-				case NotifyCollectionChangedAction.Reset:
+					foreach (INavigationControlItem elem in e.OldItems)
 					{
-						foreach (INavigationControlItem elem in getElements())
-						{
-							await AddElementToSection(elem, section);
-						}
-						foreach (INavigationControlItem elem in section.ChildItems.ToList())
-						{
-							if (!getElements().Any(x => x.Path == elem.Path))
-							{
-								section.ChildItems.Remove(elem);
-							}
-						}
-						break;
+						var match = section.ChildItems.FirstOrDefault(x => x.Path == elem.Path);
+						section.ChildItems.Remove(match);
 					}
+					if (e.Action is not NotifyCollectionChangedAction.Remove)
+						goto case NotifyCollectionChangedAction.Add;
+					break;
+				case NotifyCollectionChangedAction.Reset:
+					foreach (INavigationControlItem elem in getElements())
+						await AddElementToSection(elem, section);
+					foreach (INavigationControlItem elem in section.ChildItems.ToList())
+						if (!getElements().Any(x => x.Path == elem.Path))
+							section.ChildItems.Remove(elem);
+					break;
 			}
 		}
 
-		private bool IsLibraryOnSidebar(LibraryLocationItem item) => item is not null && !item.IsEmpty && item.IsDefaultLocation;
+		private static bool IsLibraryOnSidebar(LibraryLocationItem item)
+			=> item is not null && !item.IsEmpty && item.IsDefaultLocation;
 
 		private async Task AddElementToSection(INavigationControlItem elem, LocationItem section, int index = -1)
 		{
 			if (elem is LibraryLocationItem lib)
 			{
-				if (IsLibraryOnSidebar(lib) &&
-					await lib.CheckDefaultSaveFolderAccess() &&
-					!section.ChildItems.Any(x => x.Path == lib.Path))
+				if (IsLibraryOnSidebar(lib) && await lib.CheckDefaultSaveFolderAccess() && !section.ChildItems.Any(x => x.Path == lib.Path))
```

Comment:
A nested `if`` could be eliminated here.

Reply:
I do not agree. If elem is LibraryLocationItem but does not satisfy the other conditions, it should not pass in the last "else if" below.